primitives/state-machine/src/lib.rs

```diff
@@ -176,7 +177,7 @@ mod execution {
 	pub type DefaultHandler<E> = fn(CallResult<E>, CallResult<E>) -> CallResult<E>;
 
 	/// Trie backend with in-memory storage.
-	pub type InMemoryBackend<H> = TrieBackend<MemoryDB<H>, H>;
+	pub type InMemoryBackend<H> = TrieBackend<PrefixedMemoryDB<H>, H>;
```

Comment:
Not really related to this PR, but since there is some overhaul in progress...
Can't we define the `InMemoryBackend` in `in_memory_backend.rs` (maybe as a newtype) and replace the function `new_in_mem` with a `new` constructor in the `impl` block for that newtype?